spec/controllers/omniauth_callbacks_controller_spec.rb

```diff
@@ -15,19 +15,17 @@
                                 identifier_prefix: 'https://www.cilogon.org/')
 
     # Mock OmniAuth data for OpenID Connect with necessary info
-    OmniAuth.config.mock_auth[:openid_connect] = OmniAuth::AuthHash.new({
-                                                                          provider: 'openid_connect',
-                                                                          uid: '12345',
-                                                                          info: {
-                                                                            email: 'user@organization.ca',
-                                                                            first_name: 'Test',
-                                                                            last_name: 'User',
-                                                                            name: 'Test User'
-                                                                          }
-                                                                        })
-
     # Assign the mocked authentication hash to the request environment
-    @request.env['omniauth.auth'] = OmniAuth.config.mock_auth[:openid_connect]
+    @request.env['omniauth.auth'] = OmniAuth::AuthHash.new({
+                                                             provider: 'openid_connect',
+                                                             uid: '12345',
+                                                             info: {
+                                                               email: 'user@organization.ca',
+                                                               first_name: 'Test',
+                                                               last_name: 'User',
+                                                               name: 'Test User'
+                                                             }
+                                                           })
```

Comment:
> Its been a while since I worked on this but believe we dont need to define `@request.env['omniauth.auth'] `

I'm thinking we might need it. The `def openid_connect` method (`app/controllers/users/omniauth_callbacks_controller.rb`) has `auth = request.env['omniauth.auth']`. The tests appear to be failing without it: https://github.com/portagenetwork/roadmap/actions/runs/11409377997/job/31749474766

Rather than using `@request.env['omniauth.auth'] = OmniAuth::AuthHash.new({...` here, maybe we could  use `Rails.application.env_config['omniauth.auth'] = OmniAuth.config.mock_auth[:openid_connect]` like we did in `spec/integration/openid_connect_sso_spec.rb`?

Reply:
Also, I was conflating `@request.env` with `Rails.application.env_config`. Maybe there is a conflict occurring between these two.

Reply:
I think the real issue was that the overriding of `User.from_omniauth` in `spec/controllers/omniauth_callbacks_controller_spec.rb` was persisting beyond that test file. I've addressed the issue here: https://github.com/portagenetwork/roadmap/pull/922/commits/8d2fd866ab6245e6311a6bdf941ab1cf7043c3c8